File: appointment.py
from service import Service

class Appointment:
    def __init__(self, service=None, time=None, stylist=None, is_confirm=False, is_upcoming=True):
        self.service = service  # Service object
        self.time = time  # Appointment time
        self.stylist = stylist  # Stylist name
        self.is_confirm = is_confirm  # Boolean to confirm appointment
        self.is_upcoming = is_upcoming  # Boolean to check if appointment is upcoming

    def update_appointment_service(self, service_name, description, price):
        new_service = Service()
        if new_service.add_service(service_name, description, price):
            self.service = new_service
        else:
            raise ValueError("Invalid service details provided.")

    def update_appointment_time(self, new_time):
        self.time = new_time

    def update_appointment_stylist(self, new_stylist):
        self.stylist = new_stylist

    def is_appointment_done(self):
        return not self.is_upcoming and self.is_confirm

    def get_service_details(self):
        if self.service:
            return {
                "name": self.service.name,
                "description": self.service.description,
                "price": self.service.price,
            }
        return None

File: app/server/src/src/appointment.ts
import { Service } from './service';

class Appointment {
    service: Service;
    time: string;
    stylist: string;
    confirm: boolean;
    upcoming: boolean;



    public get Service(): Service{
        return this.service;
    }

    public set Service(service: Service){
        this.service = service;
    }

    public get Time(): string{
        return this.time;
    }

    public set Time(time: string){
        this.time = time;
    }

    public get Stylist(): string{
        return this.stylist;
    }

    public set Stylist(stylist: string){
        this.stylist = stylist;
    }

    public get Confirm(): boolean{
        return this.confirm;
    }

    public set Confirm(confirm: boolean){
        this.confirm = confirm;
    }

    public get Upcoming(): boolean{
        return this.upcoming;
    }

    public set Upcoming(upcoming: boolean){
        this.upcoming = upcoming;
    }

    updateAppointmentService(serviceName: string, duration: number, description: string, cost: string): void {
        const newService = new Service();
        if (newService.addService(serviceName, duration, description, cost)) {
            this.service = newService;
        } else {
            throw new Error("Invalid service details provided.");
        }
    }

    updateAppointmentTime(newTime: string): void {
        this.time = newTime;
    }

    updateAppointmentStylist(newStylist: string): void {
        this.stylist = newStylist;
    }

    isAppointmentDone(): boolean {
        return !this.upcoming && this.confirm;
    }

    getServiceDetails(): { name: string; description: string; cost: string } | null {
        if (this.service) {
            return {
                name: this.service.name,
                description: this.service.description,
                cost: this.service.cost,
            };
        }
        return null;
    }

}